--- a/initd/early.c
+++ b/initd/early.c
@@ -72,7 +72,24 @@

 	early_console("/dev/console");
 
-	mount("tmpfs", "/tmp", "tmpfs", MS_NOSUID | MS_NODEV | MS_NOATIME, "mode=01777");
+	FILE *meminfo = fopen("/proc/meminfo", "r");
+	if(meminfo != NULL) {
+		unsigned long long memtotal = 0;
+		char line[256];
+		while(fgets(line, sizeof(line), meminfo)) {
+			if(sscanf(line, "MemTotal: %llu kB", &memtotal) == 1)
+			{
+				memtotal = memtotal * 1024 * 3 / 4;
+				sprintf(line, "mode=01777,size=%llu", memtotal);
+				mount("tmpfs", "/tmp", "tmpfs", MS_NOSUID | MS_NODEV | MS_NOATIME, line);
+				break;
+				fclose(meminfo);
+			}
+		}
+	}
+	else {
+		mount("tmpfs", "/tmp", "tmpfs", MS_NOSUID | MS_NODEV | MS_NOATIME, "mode=01777");
+	}
 	mkdir("/tmp/shm", 01777);
 
    mkdir("/tmp/run", 0755);
